sdk/tables/aztable/connection_string.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package aztable
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+var ErrConnectionString = errors.New("connection string is either blank or malformed")
+
+func NewTableClientFromConnectionString(tableName string, connectionString string, options *TableClientOptions) (*TableClient, error) {
+	endpoint, credential, err := parseConnectionString(connectionString)
+	if err != nil {
+		return nil, err
+	}
+	return NewTableClient(tableName, endpoint, *credential, options)
+}
+
+func NewTableServiceClientFrommConnectionString(connectionString string, options *TableClientOptions) (*TableServiceClient, error) {
+	endpoint, credential, err := parseConnectionString(connectionString)
+	if err != nil {
+		return nil, err
+	}
+	return NewTableServiceClient(endpoint, *credential, options)
+}
+
+func convertConnStrToMap(connStr string) (map[string]string, error) {
+	ret := make(map[string]string)
+	connStr = strings.TrimRight(connStr, ";")
+
+	splitString := strings.Split(connStr, ";")
+	if len(splitString) == 0 {
+		return ret, ErrConnectionString
+	}
+	for _, stringPart := range splitString {
+		parts := strings.Split(stringPart, "=")
+		if len(parts) != 2 {
+			return ret, ErrConnectionString
+		}
+		ret[parts[0]] = parts[1]
+	}
+	return ret, nil
+}
+
+func parseConnectionString(connStr string) (string, *azcore.Credential, error) {
```

Comment:
Even though it's a private method it'd be nice to document it.

For instance, it's not immediately intuitive what that first 'string' is in the return value.

I don't know if it's frowned upon, but using named return values would also do the trick here. (@JeffreyRichter , @jhendrixMSFT?)